File: src/compiler/interpreter/OperatorTypeException.java
package compiler.interpreter;

import compiler.lexer.token.OperatorToken;
import compiler.lexer.token.Token;
import compiler.parser.UserException;

public class OperatorTypeException extends UserException {

  public OperatorTypeException(OperatorToken operator, Token left, Token right) {
    super(formatMessage(operator, left, right), right);
  }

  private static String formatMessage(OperatorToken operator, Token left, Token right) {
    return "Operator '" + operator.getValue() + "' is not defined between types " + left.getValue() + " and " + right.getValue();
  }

}

File: src/compiler/interpreter/OperatorTypeException.kt
package compiler.interpreter

import compiler.parser.Language.Token
import compiler.parser.Language.Token.OperatorToken
import compiler.parser.UserException

class OperatorTypeException(op: OperatorToken, left: Token, right: Token) :
        UserException("Operator '${op.str}' is not defined between types ${left.str} and ${right.str}", right)
